array.c

```diff
@@ -7896,6 +7896,45 @@ rb_ary_sum(int argc, VALUE *argv, VALUE ary)
     return v;
 }
 
+/*
+ * call-seq:
+ *   array.average() -> object
+ *   array.average() {|element| ... } -> object
+ *
+ *  When no block is given, returns the object equivalent to:
+ *    sum = 0
+ *    array.each {|element| sum += element }
+ *    sum / size
+ *  For example, <tt>[e1, e2, e3].average</tt> returns <tt>(e1 + e2 + e3) / 3</tt>.
+ *
+ *  Examples:
+ *    a = [0, 1, 2, 3]
+ *    a.average # => 1.5
+ *
+ *  When a block is given, it is called with each element
+ *  and the block's return value (instead of the element itself) is used as the addend:
+ *    a = [-3, -2, -1]
+ *    s = a.average { |element| element.abs }
+ *    s # => 2
+ *
+ */
+
+static VALUE
+rb_ary_average(int argc, VALUE *argv, VALUE ary)
+{
+    long len = RARRAY_LEN(ary);
+
+    if (len == 0)
+        return 0;
```

Comment:
This equals `Qfalse`.
Does it mean that the result is N/A?

Reply:
@nobu eeek... I am so sorry for my poor C skills... 

```ruby
[].average # NaN
```

The empty array should return `NaN` for its average value, we are calculating `sum of array / length of array`, `0 / 0`.